diff --git a/components/content_settings/core/browser/content_settings_default_provider.cc b/components/content_settings/core/browser/content_settings_default_provider.cc
index b4ff0f8..627c907 100644
--- a/components/content_settings/core/browser/content_settings_default_provider.cc
+++ b/components/content_settings/core/browser/content_settings_default_provider.cc
@@ -90,6 +90,14 @@ constexpr char kBug364820109AlreadyWorkedAroundPref[] =
     "profile.did_work_around_bug_364820109_default";
 #endif  // !BUILDFLAG(IS_IOS)
 
+
+#if BUILDFLAG(IS_LINUX)
+constexpr char kDownstreamJavascriptJitSettingToClear[] =
+    "profile.default_content_setting_values.javascript_jit";
+constexpr char kDownstreamJavascriptJitSettingAlreadyWorkedAroundPref[] =
+    "profile.did_work_around_downstream_javascript_jit_default";
+#endif  // BUILDFLAG(IS_LINUX)
+
 base::Value GetDefaultValue(const WebsiteSettingsInfo* info) {
   const base::Value& initial_default = info->initial_default_value();
   if (initial_default.is_none())
@@ -158,6 +166,12 @@ void DefaultProvider::RegisterProfilePrefs(
   // TODO(https://crbug.com/367181093): clean this up.
   registry->RegisterBooleanPref(kBug364820109AlreadyWorkedAroundPref, false);
 #endif  // !BUILDFLAG(IS_IOS)
+
+
+#if BUILDFLAG(IS_LINUX)
+  registry->RegisterBooleanPref(
+      kDownstreamJavascriptJitSettingAlreadyWorkedAroundPref, false);
+#endif  // BUILDFLAG(IS_LINUX)
 }
 
 DefaultProvider::DefaultProvider(PrefService* prefs,
@@ -413,6 +427,14 @@ void DefaultProvider::DiscardOrMigrateObsoletePreferences() {
   // TODO(https://crbug.com/367181093): clean this up.
   prefs_->ClearPref(kBug364820109AlreadyWorkedAroundPref);
 #endif  // !BUILDFLAG(IS_IOS)
+
+
+#if BUILDFLAG(IS_LINUX)
+  if (!prefs_->GetBoolean(kDownstreamJavascriptJitSettingAlreadyWorkedAroundPref)) {
+    prefs_->ClearPref(kDownstreamJavascriptJitSettingToClear);
+    prefs_->SetBoolean(kDownstreamJavascriptJitSettingAlreadyWorkedAroundPref, true);
+  }
+#endif  // BUILDFLAG(IS_LINUX)
 }
 
 void DefaultProvider::MigrateGeolocationDefaultValue() {
diff --git a/components/content_settings/core/browser/content_settings_pref_provider.cc b/components/content_settings/core/browser/content_settings_pref_provider.cc
index ac1fd5c..2889268 100644
--- a/components/content_settings/core/browser/content_settings_pref_provider.cc
+++ b/components/content_settings/core/browser/content_settings_pref_provider.cc
@@ -79,6 +79,14 @@ constexpr char kBug364820109AlreadyWorkedAroundPref[] =
     "profile.did_work_around_bug_364820109_exceptions";
 #endif  // !BUILDFLAG(IS_IOS)
 
+
+#if BUILDFLAG(IS_LINUX)
+constexpr char kDownstreamJavascriptJitSettingToClear[] =
+    "profile.content_settings.exceptions.javascript_jit";
+constexpr char kDownstreamJavascriptJitSettingAlreadyWorkedAroundPref[] =
+    "profile.did_work_around_downstream_javascript_jit_exceptions";
+#endif  // BUILDFLAG(IS_LINUX)
+
 }  // namespace
 
 // ////////////////////////////////////////////////////////////////////////////
@@ -126,6 +134,12 @@ void PrefProvider::RegisterProfilePrefs(
   // TODO(https://crbug.com/367181093): clean this up.
   registry->RegisterBooleanPref(kBug364820109AlreadyWorkedAroundPref, false);
 #endif  // !BUILDFLAG(IS_IOS)
+
+
+#if BUILDFLAG(IS_LINUX)
+  registry->RegisterBooleanPref(
+      kDownstreamJavascriptJitSettingAlreadyWorkedAroundPref, false);
+#endif  // BUILDFLAG(IS_LINUX)
 }
 
 PrefProvider::PrefProvider(PrefService* prefs,
@@ -453,6 +467,14 @@ void PrefProvider::DiscardOrMigrateObsoletePreferences() {
   // TODO(https://crbug.com/367181093): clean this up.
   prefs_->ClearPref(kBug364820109AlreadyWorkedAroundPref);
 #endif  // !BUILDFLAG(IS_IOS)
+
+
+#if BUILDFLAG(IS_LINUX)
+  if (!prefs_->GetBoolean(kDownstreamJavascriptJitSettingAlreadyWorkedAroundPref)) {
+    prefs_->ClearPref(kDownstreamJavascriptJitSettingToClear);
+    prefs_->SetBoolean(kDownstreamJavascriptJitSettingAlreadyWorkedAroundPref, true);
+  }
+#endif  // BUILDFLAG(IS_LINUX)
 }
 
 void PrefProvider::MigrateGeolocationExceptions() {
